### PR TITLE
BaseButton should respect the isBeakVisible from menuProps

### PR DESCRIPTION
**Pull request by christiango**
… menu props.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file if publishing 
- [x] New feature, bugfix, or **enhancement**
  - [ ] Includes tests
- [ ] Documentation update

#### Description of changes
Buttons render a contextual menu on click when menu props are passed in. The current implementation does not respect isBeakVisible when it is set on the menu props.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 1]
```diff
@@ -278,7 +278,7 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
     return (
       <ContextualMenu
         className={ css('ms-BaseButton-menuHost') }
-        isBeakVisible={ true }
+        isBeakVisible={ typeof menuProps.isBeakVisible === 'undefined' ? true : menuProps.isBeakVisible }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: it is a bit weird that we aren't spreading menuprops here... `{ ...menuProps }` and then layering the target on that. And css wrapping a string is useless... and menuProps.className is getting ignored, as is `data-automation-id`.

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Could you just spread the whole object, and provide only the specific overrides after?

E.g.

```
<ContextualMenu
  {...menuProps }
  className={ css('ms-BaseButton-menuhost', menuProps.className) }
  target={ this._buttonElement }
/>

There is no reason to override the default directionalHint or manually pass in items.

Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Agreed, I'll update it accordingly

File: packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx [hunk 2]
```diff
@@ -277,10 +277,10 @@ export class BaseButton extends BaseComponent<IBaseButtonProps, IBaseButtonState
   private _onRenderMenu(menuProps: IContextualMenuProps): JSX.Element {
     return (
       <ContextualMenu
-        className={ css('ms-BaseButton-menuHost') }
-        isBeakVisible={ true }
-        directionalHint={ DirectionalHint.bottomLeftEdge }
-        items={ menuProps.items }
+        {...menuProps}
+        className={ css('ms-BaseButton-menuhost', menuProps.className) }
+        isBeakVisible={ typeof menuProps.isBeakVisible === 'undefined' ? true : menuProps.isBeakVisible }
```
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: maybe a better fix would just be to move this before the mixin :)

```
isBeakVisible={ true }
anythingThatShouldBeOverridable={ value }
{ ...menuProps }
anythingThat ShouldBeForced={ value }
target={ this._buttonElement }

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: That's how we've done this in other situations.  defaults/overridable -> menuProps -> buttonProps/set in stone

Comment (christiango) [inline packages/office-ui-fabric-react/src/components/Button/BaseButton.tsx]: Yeah, it's obvious in hindsight. Thanks!

**Comment by christiango** [comment]
@dzearing anything else you want me to address here?

**Comment by micahgodbolt** [comment]
think we're good here.

**Pull request merged**